js/interface/add-markers.js

```diff
@@ -335,8 +337,53 @@ Fliplet.InteractiveMap.component('add-markers', {
         return
       }
 
-      this.showEditMarkerOverlay = !this.showEditMarkerOverlay;
-      this.reloadData();
+      if (!this.showEditMarkerOverlay) {
+        this.allMarkerStyles.forEach(elem => {
+          this.styleNames.push({'oldStyleName' : elem.name, 'newStyleName' : undefined})
+        })
+        Fliplet.Widget.toggleSaveButton(false)
+      }
+
+      if (this.showEditMarkerOverlay) {
+        $('.marker-overlay-saving').removeClass('hidden')
+
+        for (var i = 0; i < this.styleNames.length; i++) {
+          this.styleNames[i].newStyleName = this.allMarkerStyles[i].name
+        }
+
+        Fliplet.DataSources.connect(this.dataSourceId).then(connection => {
+          this.dataSourceConnection = connection
+
+          connection.find().then(records => {
+            if (!records.length) {
+              return
+            }
+
+            records.forEach(elem => {
+              var matchedStyleName = _.find(this.styleNames, function(style) {
```

Comment:
`var` is not allowed when using code that gets transpiled

Reply:
fixed